Mathlib/LinearAlgebra/Pi.lean

```diff
@@ -105,6 +105,11 @@ theorem iInf_ker_proj : (⨅ i, ker (proj i : ((i : ι) → φ i) →ₗ[R] φ i
       exact (mem_bot _).2 (funext fun i => h i)
 #align linear_map.infi_ker_proj LinearMap.iInf_ker_proj
 
+instance CompatibleSMul.pi (R S M N ι : Type*) [Semiring S]
+    [AddCommMonoid M] [AddCommMonoid N] [SMul R M] [SMul R N] [Module S M] [Module S N]
+    [LinearMap.CompatibleSMul M N R S] : LinearMap.CompatibleSMul M (ι → N) R S where
+  map_smul f r m := by ext i; apply ((LinearMap.proj i).comp f).map_smul_of_tower
+
```

Comment:
I'm somewhat skeptical of the value of these instances, or even of the `LinearMap.CompatibleSMul` typeclass itself. Are they actually needed by this PR?

Reply:
It's needed in (the generalization of) `Basis.smul`, which depends on `Basis.isScalarTower_finsupp`, which depends on `isScalarTower_of_injective`, which consumes a `LinearMap.CompatibleSMul` instance that is supplied by `LinearMap.CompatibleSMul.finsupp_dom` together with `IsScalarTower.compatibleSMul'`. `finsupp_cod` and this one (`.pi`) are not used, but I think there is no harm according to the benchmark. Probably there isn't much (or any) instances that consumes CompatibleSMul but doesn't also output CompatibleSMul?